packages/labs/src/inputList.tsx

```diff
@@ -0,0 +1,293 @@
+     * Callback invoked when an item from the list is selected,
+     * typically by clicking or pressing `enter` key.
+     */
+    onItemSelect: (item: T, event: React.SyntheticEvent<HTMLElement>) => void;
+}
+
+export interface IInputListProps<T> extends IListItemsProps<T> {
+    /**
+     * The active item is the current keyboard-focused element.
+     * Listen to `onActiveItemChange` for updates from interactions.
+     */
+    activeItem: T;
+
+    /**
+     * Invoked when user interaction should change the active item: arrow keys move it up/down
+     * in the list, selecting an item makes it active, and changing the query may reset it to
+     * the first item in the list if it no longer matches the filter.
+     */
+    onActiveItemChange: (activeItem: T) => void;
+
+    /**
+     * Callback invoked when user presses a key, after processing `InputList`'s own key events
+     * (up/down to navigate active item). This callback is passed to `compose` and (along with
+     * `onKeyUp`) can be attached to arbitrary content elements to support keyboard selection.
+     */
+    onKeyDown?: React.KeyboardEventHandler<HTMLElement>;
+
+    /**
+     * Callback invoked when user releases a key, after processing `InputList`'s own key events
+     * (enter to select active item). This callback is passed to `compose` and (along with
+     * `onKeyDown`) can be attached to arbitrary content elements to support keyboard selection.
+     */
+    onKeyUp?: React.KeyboardEventHandler<HTMLElement>;
+
+    /**
+     * Customize rendering of the input list.
+     * Receives an object with props that should be applied to elements as necessary.
+     */
+    renderer: (listProps: IInputListRendererProps<T>) => JSX.Element;
+
+    /**
+     * Query string passed to `itemListPredicate` or `itemPredicate` to filter items.
+     * This value is controlled: its state must be managed externally by attaching an `onChange`
+     * handler to the relevant element in your `renderer` implementation.
+     */
+    query: string;
+}
+
+/**
+ * An object with the following properties will be passed to an `InputList` `renderer`.
+ * Required properties will always be defined;  optional ones will only be defined if
+ * they are passed as props to the `InputList`.
+ *
+ * This interface is generic, accepting a type parameter `<T>` for an item in the list.
+ */
+export interface IInputListRendererProps<T> extends IProps {
+    /** The item focused by the keyboard (arrow keys). This item should stand out visually from the rest. */
+    activeItem: T;
+
+    /**
+     * Array of filtered items from the list (those that matched the predicate with the current `query`).
+     * See `items` for the full unfiltered list.
+     */
+    filteredItems: T[];
+
+    /**
+     * Selection handler that should be invoked when a new item has been chosen,
+     * perhaps because the user clicked it.
+     */
+    handleItemSelect: (item: T, event: React.SyntheticEvent<HTMLElement>) => void;
```

Comment:
this is not an event listener prop for you to supply but an actual handler function, passed to the `renderer` callback prop:

```jsx
// very pseudocode ahead
<InputList renderer=({ handleItemSelect }) => <div onClick={handleItemSelect} />) />
```